### ISSUE TITLE
Camel k 1.12.0: camel:jta not found in Camel catalog

### ISSUE THREAD

**Issue by logabstract** [apache/camel-k#4361]
Hello,

We are migrating some services from Camel K 1.10.2 to Camel K 1.12.0, and I have encountered a problem with the camel-jta dependency. I tested it in a minimal example:

``` yaml
# camel-k: language=yaml
# camel-k: dependency=camel-jta

- route:
    id: main
    from:
      uri: timer:tick
      parameters:
        repeatCount: 1
      steps:
        - log: hello world
```

This integration runs without problem in Camel K 1.10.2. However, in Camel K 1.12.0, the following error shows:

```
kamel run main.yaml --dev
Modeline options have been loaded from source files
Full command: kamel run main.yaml --dev --dependency=camel-jta
Warning: dependency camel:jta not found in Camel catalog
Integration "main" created
Progress: integration "main" in phase Initialization
Progress: integration "main" in phase Error
Error: integration "main" deployment failed
```

On the Camel Quarkus website, it shows that this dependency should work in both Camel Quarkus 2.13 (camel-k 1.10.2) and Camel Quarkus 2.16 (camel-k 1.12.0):
https://camel.apache.org/camel-quarkus/2.13.x/reference/extensions/jta.html

Link to zulip post: [link](https://camel.zulipchat.com/#narrow/stream/257299-camel-k/topic/camel.3Ajta.20not.20found.20in.20Camel.20catalog)

**Comment by squakez** [comment]
The component is missing from the catalog.

@christophd I think you had a look at some similar problem recently. Probably it depends on Camel Quarkus platform, can you confirm?

**Comment by lfabriko** [comment]
Passing info from @christophd :
Problem caused by https://github.com/apache/camel-k/commit/2da65f2681f67904cfc719deb6573286be8ce92c: "Root cause is that camel-jta for some reason is not in the catalog and that we have added the validation check to raise errors if the component could not be found in the catalog in 1.12."

Reproduced:
- minimal example integration runs correctly on `v1.10.2`
- fails on `v1.12.0` for both `camel-jta` and `camel:jta`
- runs correctly on `v1.12.0` after removing commit ^ https://github.com/apache/camel-k/commit/2da65f2681f67904cfc719deb6573286be8ce92c

**Comment by christophd** [comment]
also, we found out that `camel-jta` has never been in the catalog. it has been working with 1.10 because the explicit catalog check from [2da65f2](https://github.com/apache/camel-k/commit/2da65f2681f67904cfc719deb6573286be8ce92c) is missing in 1.10

I think we should also open a new issue in [camel-quarkus](https://github.com/apache/camel-quarkus) to find out why `camel-jta` is missing in the catalog

**Comment by christophd** [comment]
There must be another mechanism that resolves dependencies besides that one working on the catalog because in 1.10.2 the integration is working ok despite the fact that the `camel-jta` is also not part of the catalog in this case.

In 1.12 it fails fast due to the check on the missing component in the catalog

**Comment by squakez** [comment]
Fixed by #4402

**Comment by christophd** [comment]
Also fixed by https://github.com/apache/camel-k-runtime/issues/1029
